### PR TITLE
docs: Update common readme

### PR DESCRIPTION
**Pull request by NicholasBoll**

### CODE REVIEW

**Review by mannycarrera4 — APPROVED** [review]
thanks for adding this!

**Review by alanbsmith — APPROVED** [review]
Thanks for the update!

File: modules/react/common/README.md [hunk 1]
```diff
@@ -289,6 +289,10 @@ only provide a common model for the subcomponents.
 
 ### `ExtractProps`
 
+v5 introduces components that could take an `as` prop, which means the underlying element that is
+rendered can change. Components props that use the `as` prop no longer extend an HTML attribute
```
Comment (alanbsmith) [inline modules/react/common/README.md:293]: Whoops. Missed this.
```suggestion
rendered can change. Component props that use the `as` prop no longer extend an HTML attribute
```

File: modules/react/common/README.md [hunk 2]
```diff
@@ -337,11 +341,9 @@ this example, `onClick` doesn't exist on `MyComponent`, but since `MyComponent`
 If we inspect the `MyNewComponentProps` type, it will return the following:
 
 ```tsx
-React.ClassAttributes<HTMLElement> & React.HTMLAttributes<HTMLElement> & MyComponentProps
+React.HTMLAttributes<HTMLElement> & MyComponentProps
```
Comment (NicholasBoll) [inline modules/react/common/README.md:344]: The order was changed to list props first, then HTML attributes

```suggestion
MyComponentProps & React.HTMLAttributes<HTMLElement>
```

**Pull request merged**